tests/functional/test_freeze.py

```diff
@@ -652,3 +652,52 @@ def test_freeze_user(script, virtualenv, data):
         <BLANKLINE>""")
     _check_output(result.stdout, expected)
     assert 'simple2' not in result.stdout
+
+
+def test_freeze_path(tmpdir, script, data):
+    """
+    Test freeze with --path.
+    """
+    script.pip('install', '--find-links', data.find_links,
+               '--target', tmpdir, 'simple==2.0')
+    result = script.pip('freeze', '--path', tmpdir)
+    expected = textwrap.dedent("""\
+        simple==2.0
+        <BLANKLINE>""")
+    _check_output(result.stdout, expected)
+
+
+def test_freeze_path_exclude_user(tmpdir, script, data):
+    """
+    Test freeze with --path and make sure packages from --user are not picked
+    up.
+    """
+    script.pip_install_local('--find-links', data.find_links,
+                             '--user', 'simple2')
+    script.pip('install', '--find-links', data.find_links,
+               '--target', tmpdir, 'simple==2.0')
+    result = script.pip('freeze', '--path', tmpdir)
```

Comment:
It seems like it would help to have a "before" case here in addition to the "after" to make sure this test is testing what it says it's testing. In other words, first call `pip freeze` with the `--user` flag to confirm that a certain string is present. And then when you call with the `--path` flag, you can check that that string isn't present. Does that seem reasonable?

Reply:
I modified this to install the `simple2` package with `--user`, then validate it with `freeze --user` before performing the `--target` installation and checking the installation with `freeze --path`.

Reply:
For this suggestion, what I meant was calling `pip freeze` with the `--user` flag after the last install invocation but before `pip freeze --path`. That way you're checking the two outputs against the same state (so you know the difference in the two outputs is due to the arguments rather than the last install invocation). Does that make sense?